OneCommunityGlobal/HighestGoodNetworkApp
# Olena_edit_delete_functionality_for_lesson_card

**Description**
Lesson list component for Phase II
<img width="680" alt="lessonlistpage" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49422649/28583fa9-f2fe-45f0-b12d-1051b1137a99">


**Related PRs (if any):**
#1479
…

**Main changes explained:**
Implemented functionality for the Edit & Delete buttons on the Lesson Card (visible and operable only for the author of the card).
Users can click on the Edit button to modify the lesson summary, with added validation to ensure the summary is not empty.
Users can click on the Delete button to remove the entire card.



**How to test:**

1. Checkout to the current branch.
2. Run npm install and ... to run this PR locally.
3. Log in as any user.
4. Go to http://localhost:3000/bmdashboard/lessonlist.
5. Verify the design, tag input functionality, and expand/collapse card functionality (PR1479).
6. Test the Edit & Delete buttons. To test, please change the userId to your userID in DummyData (src -> components -> BMDashboard -> LessonList -> Lessons.jsx -> DummyData -> userid).

**## Note:**
Please note that the Figma navigation bar and the current design differ. I've made adjustments based on Jae's request, and he has also verified the design.
Currently, I am using dummy data, and the frontend is not connected to the backend. LocalStorage is not utilized to store data, so the functionality is not persistent after page refresh. I'll connect all these features to the backend and will load data from DB later, as the schema, routers and controllers are still in progress.



## Screenshots or videos of changes:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49422649/fe50da50-953d-47ce-bdaf-77187d05b9e7
